src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,965 @@
+            }
+
+            //The amount of numbers to the next turn increases every other turn.
+            this.numberToTurnAtForSpiral = 1
+            this.incrementForNumberToTurnAt = 1
+            this.whetherIncrementShouldIncrement = true
+        }
+    }
+
+    fillGridCell() {
+        this.drawBackgroundSquare()
+
+        this.drawPrimaryColorSquare()
+
+        this.drawSecondaryColorSquare()
+
+        this.showNumber()
+    }
+
+    drawBackgroundSquare() {
+        this.sketch.fill(this.backgroundColor)
+        this.drawBigSquare()
+    }
+
+    drawPrimaryColorSquare() {
+        this.colorMainColorProperties()
+        this.drawBigSquare()
+    }
+
+    drawSecondaryColorSquare() {
+        this.colorSecondaryColorProperties()
+        this.drawSmallSquare()
+    }
+
+    colorMainColorProperties() {
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization
+                    === PropertyVisualization.Primary_Color
+            ) {
+                if (
+                    this.hasProperty(
+                        this.currentNumber,
+                        this.propertyObjects[i].property
+                    )
+                ) {
+                    this.sketch.fill(this.propertyObjects[i].color)
+                }
+            }
+        }
+    }
+
+    colorSecondaryColorProperties() {
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization
+                    === PropertyVisualization.Secondary_Color
+            ) {
+                if (
+                    this.hasProperty(
+                        this.currentNumber,
+                        this.propertyObjects[i].property
+                    )
+                ) {
+                    this.sketch.fill(this.propertyObjects[i].color)
+                }
+            }
+        }
+    }
+
+    drawBigSquare() {
+        this.sketch.rect(
+            this.x,
+            this.y,
+            this.scalingFactor,
+            this.scalingFactor
+        )
+    }
+
+    drawSmallSquare() {
+        this.sketch.rect(
+            this.x + this.scalingFactor / 4,
+            this.y + this.scalingFactor / 4,
+            this.scalingFactor / 2,
+            this.scalingFactor / 2
+        )
+    }
+
+    hasProperty(num: bigint, property: Property) {
+        if (property === Property.Number_Sequence) {
+            for (
+                let a = this.seq.first;
+                a < Math.min(this.seq.last - this.seq.first + 1, 200);
+                a++
+            ) {
+                if (this.seq.getElement(a) == num) {
+                    return true
+                }
+            }
+            return false
+        } else if (property === Property.Prime) {
```

Comment:
I'm trying to understand the big picture on Property.Number_Sequence.  Is this only for presets comparing two sequences?  It looks as if it detects whether num is in this.seq.  But unless we are in such a preset, isn't this always true, because we are only calling hasProperty() on nums from this.seq to begin with?  At least this is true I think if the user chose to use Number_Sequence as a property explicitly.  (Let me know if I'm misunderstanding of course.) Perhaps this capability is only here for use with the presets that compare two sequences, in which case the user shouldn't be able to choose Number_Sequence in the dialog at all, because it doesn't make sense outside the presets.  I wasn't able to use that setting in the dialog and produce any output (seemed to get stuck in an infinite loop).

Reply:
For this PR, we are going to temporarily disable anything having to do with multiple sequences.